tests/test_rff.py

```diff
@@ -0,0 +1,538 @@
+    alpha["rff_sp"] = 1234
+    out_expected = (
+        alpha.set_index(["model", "ssp", "rff_sp", "year"])
+        .drop(columns="var")
+        .to_xarray()["value"]
+    )
+    out_actual = clean_simulation(1234, str(Path(tmp_path) / "clean_root"))
+
+    xr.testing.assert_equal(out_actual, out_expected)
+
+
+def test_clean_weights_error(tmp_path, weights_unclean):
+    """
+    Test that clean weights correctly cleans alpha and error out of the weights file
+    """
+    error = list(product(np.arange(2010, 2021, 5), ["USA", "ARG"], ["error"], [1]))
+    error = pd.DataFrame(error, columns=["year", "iso", "var", "value"])
+    error["rff_sp"] = 1234
+    out_expected = (
+        error.set_index(["iso", "year", "rff_sp"])
+        .drop(columns="var")
+        .to_xarray()["value"]
+    )
+    out_actual = clean_error(1234, str(Path(tmp_path) / "clean_root"))
+
+    xr.testing.assert_equal(out_actual, out_expected)
+
+
+def test_weight_df(tmp_path):
+    """
+    Test that weight_df correctly weights SSP damage functions to produce RFF damage functions
+    """
+    d = tmp_path / "weighting"
+    d.mkdir()
+
+    sector = "sector_dummy"
+    pulse_year = "2020"
+    recipe = "euler_ramsey"
+    disc = "risk_aversion"
+    eta_rho = [1, 2]
+    file = "damage_functions"
+    mask = "unmasked"
+
+    df_in_dir = d / sector / pulse_year / mask
+    df_out_dir = d / "out" / sector / pulse_year / mask
+
+    df_in_file = (
+        df_in_dir / f"{recipe}_{disc}_eta{eta_rho[0]}_rho{eta_rho[1]}_{file}.nc4"
+    )
+    outfile_path = (
+        df_out_dir / f"{recipe}_{disc}_eta{eta_rho[0]}_rho{eta_rho[1]}_{file}.nc4"
+    )
+    fractional_path = (
+        df_out_dir
+        / f"{recipe}_{disc}_eta{eta_rho[0]}_rho{eta_rho[1]}_fractional_{file}.nc4"
+    )
+
+    Path(df_in_dir).mkdir(parents=True, exist_ok=True)
+    Path(df_out_dir).mkdir(parents=True, exist_ok=True)
+    df_in = xr.Dataset(
+        {
+            "anomaly": (
+                ["discount_type", "ssp", "model", "year"],
+                np.ones((1, 2, 2, 3)),
+            ),
+        },
+        coords={
+            "discount_type": (["discount_type"], ["euler_ramsey"]),
+            "ssp": (["ssp"], ["SSP3", "SSP4"]),
+            "model": (["model"], ["IIASA GDP", "OECD Env-Growth"]),
+            "year": (["year"], [2021, 2022, 2099]),
+        },
+    )
+    df_in.to_netcdf(df_in_file)
+
+    factors = xr.Dataset(
+        {
+            "anomaly": (
+                ["discount_type", "ssp", "model", "year"],
+                np.ones((1, 2, 2, 3)),
+            ),
+        },
+        coords={
+            "discount_type": (["discount_type"], ["euler_ramsey"]),
+            "ssp": (["ssp"], ["SSP3", "SSP4"]),
+            "model": (["model"], ["IIASA GDP", "OECD Env-Growth"]),
+            "year": (["year"], [2100, 2101, 2102]),
+        },
+    )
+    df_in.to_netcdf(df_in_file)
+
+    out_expected = xr.Dataset(
```

Comment:
I think this test in particular could use a few inline comments to explain what is what. For example, what are `factors` meant to be? Why does `out_expected` have ssp and model dims but `out_expected_fractional` not? I thought all outputs would be RFF-based

Reply:
Ah this is because I mistakenly weighted with the wrong dimensions (ssp instead of runid). Will fix + comment.